crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2704,7 +2704,80 @@ impl<'db> TypeInferenceBuilder<'db> {
             (_, Type::BytesLiteral(_)) => {
                 self.infer_binary_type_comparison(left, op, KnownClass::Bytes.to_instance(self.db))
             }
+            (Type::Tuple(lhs), Type::Tuple(rhs)) => {
+                // Note: This only works on heterogeneous tuple types.
+                let lhs_elements = lhs.elements(self.db).as_ref();
+                let rhs_elements = rhs.elements(self.db).as_ref();
 
+                match op {
+                    ast::CmpOp::Eq
+                    | ast::CmpOp::NotEq
+                    | ast::CmpOp::Lt
+                    | ast::CmpOp::LtE
+                    | ast::CmpOp::Gt
+                    | ast::CmpOp::GtE => {
+                        self.infer_lexicographic_type_comparison(lhs_elements, op, rhs_elements)
+                    }
+                    ast::CmpOp::In | ast::CmpOp::NotIn => {
+                        let mut eq_cnt = 0usize;
+                        let mut not_eq_cnt = 0usize;
```

Comment:
As a general principle, I prefer to use full words in variable names -- and in this specific case I would somewhat strongly prefer to use "count" for these names